lib/routes/account.js

```diff
@@ -333,6 +349,135 @@ module.exports = function (
+                      emailCode: crypto.randomBytes(16),
+                      emailVerified: true,
+                      kA: kA,
+                      wrapWrapKb: crypto.randomBytes(32),
+                      accountResetToken: null,
+                      passwordForgotToken: null,
+                      authSalt: authSalt,
+                      verifierVersion: 0,
+                      verifyHash: crypto.randomBytes(32),
+                      openId: id,
+                      verifierSetAt: Date.now(),
+                      locale: locale
+                    }
+                  )
+                }
+              )
+              .then(
+                function (account) {
+                  return db.createSessionToken(
+                    {
+                      uid: account.uid,
+                      email: account.email,
+                      emailCode: account.emailCode,
+                      emailVerified: true,
+                      verifierSetAt: account.verifierSetAt
+                    }
+                  )
+                  .then(
+                    function (sessionToken) {
+                      return db.createKeyFetchToken(
+                        {
+                          uid: account.uid,
+                          kA: account.kA,
+                          // wrapKb is undefined without a password.
+                          // wrapWrapKb has the properties we need for this
+                          // value; Its stable, random, and will change on
+                          // account reset.
+                          wrapKb: account.wrapWrapKb,
+                          emailVerified: true
+                        }
+                      )
+                      .then(
+                        function (keyFetchToken) {
+                          return {
+                            sessionToken: sessionToken,
+                            keyFetchToken: keyFetchToken,
+                            unwrapBKey: butil.xorBuffers(
+                              account.kA,
+                              account.wrapWrapKb
+                            )
+                            // The browser using these values for unwrapBKey
+                            // and wrapKb (from above) will yield kA
+                            // as the Sync key instead of kB
+                          }
+                        }
+                      )
+                    }
+                  )
+                  .then(
+                    function (tokens) {
+                      reply(
+                        {
+                          uid: tokens.sessionToken.uid.toString('hex'),
+                          email: account.email,
+                          session: tokens.sessionToken.data.toString('hex'),
+                          key: tokens.keyFetchToken ?
+                            tokens.keyFetchToken.data.toString('hex')
+                            : undefined,
```

Comment:
WIll `keyFetchToken` every be falsy here?  If so, we could also consider not sending `unwrap` if there's no `keyFetchToken`  since it's otherwise useless.